### PR TITLE
feat: APP-2740 - Implement EmptyStateCard Component

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description

combines Card and EmptyState into EmptyStateCard

Task: [APP-2740](https://aragonassociation.atlassian.net/browse/APP-2740)

## Type of change

-   [x] New feature (non-breaking change which adds functionality)

## Checklist:

-   [x] I have selected the correct base branch.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] Any dependent changes have been merged and published in downstream modules.
-   [x] I ran all tests with success and extended them if necessary.
-   [x] I have updated the `CHANGELOG.md` file in the root folder of the package after the [UPCOMING] title and before
        the latest version.

[APP-2740]: https://aragonassociation.atlassian.net/browse/APP-2740?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/cards/cardEmptyState/index.tsx [hunk 1]
```diff
@@ -0,0 +1 @@
+export { CardEmptyState, type CardEmptyStateProps } from './cardEmptyState';
```
Comment (Fabricevladimir) [inline src/components/cards/cardEmptyState/index.tsx:1]: Don't forget to re-export the `CardEmptyState` in the `cards` `index.ts`

**Review by Fabricevladimir — COMMENTED** [review]
Thanks! Left a couple of comments

File: src/components/cards/cardEmptyState/cardEmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { EmptyState, type IEmptyStateProps } from '../../states/emptyState';
+import { Card, type ICardProps } from '../card';
+
+export type CardEmptyStateProps = ICardProps & IEmptyStateProps;
+
+export const CardEmptyState: React.FC<CardEmptyStateProps> = (props) => {
```
Comment (RuggeroCino) [inline src/components/cards/cardEmptyState/cardEmptyState.tsx:6]: Props interfaces/types always starts with `I`
```suggestion
export const CardEmptyState: React.FC<ICardEmptyStateProps> = (props) => {
```

File: src/components/cards/cardEmptyState/cardEmptyState.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * **Note:** see <EmptyState /> for more details on implementation of layout.
+ */
+export const Default: Story = {
+    args: {
+        heading: 'Heading',
+        objectIllustration: { object: 'LIGHTBULB' },
+    },
+};
+
```
Comment (thekidnamedkd) [inline src/components/cards/cardEmptyState/cardEmptyState.stories.tsx:29]: Technically an illustration isn't required, but I think with just a heading it's going be more confusing in the documentation as to what this component is/can do in relationship to EmptyState.

File: src/components/cards/cardEmptyState/cardEmptyState.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    } else if (objectIllustration) {
+        emptyStateProps.objectIllustration = objectIllustration;
+    }
+
+    return (
+        <Card className={classNameCard}>
+            <EmptyState {...(emptyStateProps as IEmptyStateProps)} />
+        </Card>
+    );
+};
```
Comment (thekidnamedkd) [inline src/components/cards/cardEmptyState/cardEmptyState.tsx:32]: I don't love this recasting, but the discriminated API with TS is kind of particular once we go from broader structural inference.

**Review by RuggeroCino — APPROVED** [review]
LGTM, thanks @thekidnamedkd! Make sure to fix the conflicts on the changelog file before merging the PR 👌🏻

**Pull request merged**